website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+            });
+            return promise;
+        };
+
+        self.callLogs = function(filter) {
+            self.eventFilter = self.eventFilter === filter ? false : filter;
+            self.page = 1;
+            self.cache = [];
+            self.getLogs();
+        };
+
+        self.getLogs(true, false);
+
+        $(window).resize(function(){
+            var canvas = document.getElementById('rACanvas');
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+        });
+
+        self.makeLine = function(canvas){
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+            var progBar = $('#rAProgressBar');
+            var handle = $('.ui-slider-handle');
+            var leftHandle = handle[0];
+            var rightHandle = handle[1];
+            ctx.beginPath();
+            ctx.moveTo(leftHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left - $('#rACanvas').offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+            ctx.beginPath();
+            ctx.moveTo(rightHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left + progBar[0].offsetWidth - $('#rACanvas').offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+        };
+        self.categoryColor = function(category){
+            if (category.indexOf('wiki') !== -1){ return '#d9534f'; }
+            if (category.indexOf('comment') !== -1){ return '#5bc0de'; }
+            if (category.indexOf('file') !== -1){ return '#337ab7'; }
+            if (category.indexOf('project') !== -1){ return '#f0ad4e'; }
+            else { return '#5cb85c'; }
+        };
+        self.addButtons = function(ele, isInitialized) {
+            if ($('#rALeftButton')){
+                $('#rALeftButton').css('height', $('#rALogs').height());
+            }
+            if ($('#rARightButton')){
+                $('#rARightButton').css('height', $('#rALogs').height());
+            }
+        };
+        self.makeSliderProgress =  function(){
+            return '<div id="rAFilleBar" class="progress" style="height: 11px">' +
+                        '<div class="progress-bar"></div>' +
+                '</div>';
+        };
+        self.addSlider = function(ele, isInitialized){
+            var begin = (Number(self.firstDay.format('x'))/self.div | 0);
+            var end = (Number(self.today.format('x'))/self.div | 0);
+            var values = [(Number(self.dateBegin.format('x'))/self.div | 0), (Number(self.dateEnd.format('x'))/self.div | 0)];
+            var canvas = document.getElementById('rACanvas');
+            if (!isInitialized) {
+                $('#rASlider').slider({
```

Comment:
Can you cache jquery selectors within functions? It saves the program from having to traverse the DOM again to find it. 
```
var sliderElement = $('#rASlider');
```